src/ChainRouter.php

```diff
@@ -211,10 +213,34 @@ private function doMatch($pathinfo, Request $request = null)
     }
 
     /**
-     * {@inheritdoc}
-     *
      * Loops through all registered routers and returns a router if one is found.
-     * It will always return the first route generated.
+     * It will always return the first route generated, based on the given parameters.
+     *
+     * Parameters that reference placeholders in the route pattern will substitute them in the
+     * path or host. Extra params are added as query string to the URL.
+     *
+     * When the passed reference type cannot be generated for the route because it requires a different
+     * host or scheme than the current one, the method will return a more comprehensive reference
+     * that includes the required params. For example, when you call this method with $referenceType = ABSOLUTE_PATH
+     * but the route requires the https scheme whereas the current scheme is http, it will instead return an
+     * ABSOLUTE_URL with the https scheme and the current host. This makes sure the generated URL matches
+     * the route in any case.
+     *
+     * If there is no route with the given name, the generator must throw the RouteNotFoundException.
+     *
+     * The special parameter _fragment will be used as the document fragment suffixed to the final URL.
+     *
+     * @param mixed $name       the name of the route or something identifying a route that can be used by any of
+     *                          the chained routers
+     * @param mixed $parameters An array of parameters
```

Comment:
shouldn't this be `@param array $parameters`?

Reply:
makes sense, I just copied it from the UrlGeneratorInterface definition. Should I also change it there?

Reply:
Ahh, that's actually in another repository. I'll just change it here.

Reply:
can you try if `@inheritdoc` works to allow you to only specify the argument that actually is different?